src/components/sbb-expansion-panel/sbb-expansion-panel.tsx

```diff
@@ -0,0 +1,213 @@
+
+  /** Whether the panel is disabled, so its expanded state can't be changed. */
+  @Prop({ reflect: true }) public disabled = false;
+
+  /** Whether the panel has no border. */
+  @Prop({ reflect: true }) public borderless = false;
+
+  /** Whether the animations should be disabled. */
+  @Prop({ reflect: true }) public disableAnimation = false;
+
+  /** Emits whenever the sbb-expansion-panel starts the opening transition. */
+  @Event({
+    bubbles: true,
+    composed: true,
+    eventName: 'will-open',
+  })
+  public willOpen: EventEmitter<void>;
+
+  /** Emits whenever the sbb-expansion-panel is opened. */
+  @Event({
+    bubbles: true,
+    composed: true,
+    eventName: 'did-open',
+  })
+  public didOpen: EventEmitter<void>;
+
+  /** Emits whenever the sbb-expansion-panel begins the closing transition. */
+  @Event({
+    bubbles: true,
+    composed: true,
+    eventName: 'will-close',
+  })
+  public willClose: EventEmitter<void>;
+
+  /** Emits whenever the sbb-expansion-panel is closed. */
+  @Event({
+    bubbles: true,
+    composed: true,
+    eventName: 'did-close',
+  })
+  public didClose: EventEmitter<void>;
+
+  @Element() private _element: HTMLElement;
+
+  @Listen('toggle-expanded')
+  public toggleExpanded(): void {
+    this.expanded = !this.expanded;
+  }
+
+  @Watch('expanded')
+  public onExpandedChange(): void {
+    this._element
+      .querySelector('sbb-expansion-panel-header')
+      .setAttribute('aria-expanded', String(this.expanded));
+    this._element
+      .querySelector('sbb-expansion-panel-content')
+      .setAttribute('aria-hidden', String(!this.expanded));
+    this._element.style.setProperty(
+      '--sbb-expansion-panel-content-height',
+      `${this._contentElement.scrollHeight}px`,
+    );
+
+    if (this.expanded) {
+      this.willOpen.emit();
+      this.disableAnimation && this.didOpen.emit();
+    } else {
+      this.willClose.emit();
+      this.disableAnimation && this.didClose.emit();
+    }
+  }
+
+  @Watch('disabled')
+  public updateDisabledOnHeader(newDisabledValue: boolean): void {
+    this._headerRef.disabled = newDisabledValue;
+  }
+
+  private _contentElement: HTMLElement;
+  private _transitionEventController: AbortController;
+  private _progressiveId = `-${++nextId}`;
+  private _headerRef: HTMLSbbExpansionPanelHeaderElement;
+  private _contentRef: HTMLSbbExpansionPanelContentElement;
+
+  private _onHeaderSlotChange(event): void {
+    this._headerRef = (event.target as HTMLSlotElement)
+      .assignedElements()
+      .find(
+        (e): e is HTMLSbbExpansionPanelHeaderElement => e.tagName === 'SBB-EXPANSION-PANEL-HEADER',
+      );
+    if (!this._headerRef) {
+      return;
+    }
+
+    this._headerRef.setAttribute('aria-expanded', String(this.expanded));
+    if (this.disabled) {
+      this._headerRef.setAttribute('disabled', String(this.disabled));
+    }
+    if (!this._headerRef.getAttribute('id')) {
```

Comment:
Id assgnement could be moved to _linkHeaderAndContent
Nitpick; this._headerRef.id is equivalent to this._headerRef.getAttribute('id') and for me, easier to read

Reply:
Done in [23c4d25](https://github.com/lyne-design-system/lyne-components/pull/1860/commits/23c4d2557ee5eae7ed464f41ad211275b467056a)